docs/how-does-version-resolution-work-in-maven-and-gradle.md

```diff
@@ -0,0 +1,132 @@
+encounters first. Another way to describe the algorithm is that Maven chooses
+the first version encountered when traversing the dependency graph in
+breadth-first order. Looking at the same unresolved dependency graph we used
+in Gradle example 1:
+
+<img src="assets/images/ddc-resolution-06.png" alt="Maven example 1">
+
+As you can see by comparing the resolved graphs, Maven makes the opposite
+choice. Now consider at the second example, where the root has both a direct and
+indirect dependency on the same library:
+
+<img src="assets/images/ddc-resolution-07.png" alt="Maven example 2">
+
+In this case, Maven chooses A:10.0 because it is only 1 hop away from the root
+instead of 2 hops (A:10.1). In this case, Maven also makes the opposite choice
+of Gradle.
+
+It should be noted that Maven and Gradle don't always make the opposite choice.
+In the following example, the highest version is also the closest to the root,
+so Maven and Gradle both choose A:10.1.
+
+<img src="assets/images/ddc-resolution-07b.png" alt="Maven and Gradle example">
+
+## Interaction between Maven and Gradle
+
+When a library is built, its build system performs version resolution for its
+entire dependency graph. Since the build systems of the libraries in that
+dependency graph also perform their own version resolution, the versions
+selected can be different, especially when a consumer uses a different build
+system than its dependencies. Sometimes, the version selected by the consumer
+can be incompatible with the version selected by the dependency when the
+dependency is built by itself.
+
+Consider an example using Gradle example 2 as its basis, where library D depends
+on A:10.0 and C:30.0 (and thus indirectly A:10.1). Assume C depends on a feature
+added into 10.1. This means that if 10.0 is selected, C fails at runtime. From
+library D's perspective, this is fine, since Gradle chooses version 10.1.
+
+Suppose we add a new library E which uses Maven as its build system, and which
+adds D as a dependency. When E is built, Maven resolves the whole dependency
+graph, including D's subgraph, even though Gradle resolved the subgraph of D for
+when D was originally built. When Maven performs its version resolution,
+it chooses a different version of A (10.0) than Gradle did (10.1), which breaks
+C. This happens even though D works perfectly fine internally.
+
+<img src="assets/images/ddc-resolution-08.png" alt="Gradle and maven disagreement">
+
+As a consequence, the author of library E might file a bug against library
+D. The author of library D would argue that there is no bug. From their individual
+perspectives, are both right. From the ecosystem perspective, the author of
+library D unfortunately needs to adapt its dependencies so they don't cause
+problems for Maven consumers (even though they may have sworn off Maven and use
+Gradle exclusively). In this case, they have an easy fix - they can upgrade the
```

Comment:
nit: hyphen to em dash, or make this two sentences.

Reply:
Done.